.github/scripts/ci/tests/extract-exports.test.sh

```
@@ -0,0 +1,103 @@
+# Sources the REAL `extract_symbols` from extract-exports.sh and
+# feeds it synthetic TypeScript, so a regression in the gate's own
+# parser is caught here rather than by a reviewer noticing that a new
+# export shipped undocumented. Run via:
+#
+#   bash .github/scripts/ci/tests/extract-exports.test.sh
+#
+# Exit 0 = every scenario extracted the expected symbol set; 1 = a
+# scenario regressed.
+
+set -o pipefail
+cd "$(git rev-parse --show-toplevel)"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)